pkg/river/vm/vm_stdlib_test.go

```diff
@@ -0,0 +1,37 @@
+package vm_test
+
+import (
+	"testing"
+
+	"github.com/grafana/agent/pkg/river/parser"
+	"github.com/grafana/agent/pkg/river/vm"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVM_Stdlib(t *testing.T) {
+	t.Setenv("TEST_VAR", "Hello!")
+
+	tt := []struct {
+		name   string
+		input  string
+		expect interface{}
+	}{
+		{"env", `env("TEST_VAR")`, string("Hello!")},
+		{"concat", `concat([true, "foo"], [], [false, 1])`, []interface{}{true, "foo", false, 1}},
+		{"unmarshal_json object", `unmarshal_json("{\"foo\": \"bar\"}")`, map[string]interface{}{"foo": "bar"}},
+		{"unmarshal_json array", `unmarshal_json("[0, 1, 2]")`, []interface{}{float64(0), float64(1), float64(2)}},
```

Comment:
I tried toying around with these functions a little, I got some issues when decoding JSON `null`s.

When decoding them as struct fields or array elements, I got
```
		{"unmarshal_json nil field", `unmarshal_json("{\"foo\": null}")`, map[string]interface{}{"foo": nil}},
		{"unmarshal_json nil array element", `unmarshal_json("[0, null]")`, []interface{}{float64(0), nil}},

pkg/river/internal/value/decode.go|89| panic: reflect: call of reflect.Value.Type on zero Value 
```

and when trying to use concat with an array that contained a null value, I got
``` 
		{"unmarshal_json array", `concat(unmarshal_json("[0, 1, 2]"), unmarshal_json(true, null, "3"))`, []interface{}{float64(0), float64(1), float64(2), true, nil, float64(3)}},

panic: NULL is not a valid token 
```

I haven't dived in to figure why, just noting it to look into it.

Reply:
Thanks, this is likely another instance where we're not handling nils gracefully. 

The null issue might be a parser error, though that call also looks incorrect since there should only be one string argument

Reply:
I take the "NULL is not a valid token" theory back, I have no idea where that's coming from :)

Reply:
Both of these are fixed now :)